src/microbe_stage/SpawnSystem.cs

```diff
@@ -422,14 +424,14 @@ private int SpawnWithSpawner(Spawner spawnType, Vector3 location, ref int spawns
     ///   Despawns entities that are far away from the player
     /// </summary>
     /// <returns>The number of alive entities, used to limit the total</returns>
-    private int DespawnEntities(Vector3 playerPosition)
+    private float DespawnEntities(Vector3 playerPosition)
```

Comment:
Is it *really* necessary to weight the despawning as well? Does it actually take that much more time to despawn a big cell as a small one? Also now with the adjusted despawn amounts, this can despawn a lot more chunks per frame (which may increase game instability).

Reply:
I don't know to be honest. `estimateEntityCount` takes its value from this method each frame before spawning happens, which is what suggested to me that it should be a float.

Reply:
I commented on this line, but my comment was more general. In my opinion the game crashing etc. unstability can be caused by how many things are despawned, so I'd prefer to use an int limit for despawning, but this method should still return a float with the currently alive weighted entity count.

Reply:
I've changed this and the other comment I still had. I think I may have also fixed #3686 as well as I noticed a problem with the break behaviour in the despawn loop.